crates/uv-resolver/src/resolution.rs

```diff
@@ -376,21 +418,30 @@ impl std::fmt::Display for DisplayResolutionGraph<'_> {
                     .collect::<Vec<_>>();
                 edges.sort_unstable_by_key(|package| package.name());
 
-                match edges.len() {
-                    0 => {}
-                    1 => {
-                        for dependency in edges {
-                            writeln!(f, "{}", format!("    # via {}", dependency.name()).green())?;
+                match self.annotation_style {
+                    AnnotationStyle::Line => {
+                        if !edges.is_empty() {
+                            sep = if self.show_hashes { "\n    " } else { "  " };
+                            annotation = DisplayResolutionGraph::annotation_line(edges);
                         }
                     }
-                    _ => {
-                        writeln!(f, "{}", "    # via".green())?;
-                        for dependency in edges {
-                            writeln!(f, "{}", format!("    #   {}", dependency.name()).green())?;
+                    AnnotationStyle::Split => {
+                        if !edges.is_empty() {
+                            sep = "\n";
                         }
+                        annotation = DisplayResolutionGraph::annotation_split(edges);
                     }
                 }
             }
+            // Assemble the line with the annotations and remove trailing whitespaces.
+            line = format!("{line:24}{sep}{annotation}")
```

Comment:
Can you walk me through `line:24` here?

Reply:
Sure! I'm trying to keep the same logic, as `pip-tools`, to minimize the diff in the output file. The align `24` is something they use in the `pip-tools` as a reasonable number to align the annotation comments: https://github.com/jazzband/pip-tools/blob/main/piptools/writer.py#L318 and I agree with them. Most of the times annotation would be aligned except some extra-long dependencies.

```requirements.txt
monotonic==1.6            # via analytics-python
oauthlib==2.1.0           # via pyproject.toml
opentelemetry-api==1.22.0  # via ddtrace
packaging==21.3           # via gunicorn, pyproject.toml
prompt-toolkit==3.0.43    # via click-repl
protobuf==4.25.3          # via ddsketch, ddtrace
```

Reply:
Do you mean to extract `annotation_line` and `annotation_split` onto the `match` statement? I could do this, they're not very big, should be fine

Reply:
Yeah, you're right, it's easier to read without those function calls :)